lib/acts_as_ar_query.rb

```diff
@@ -194,6 +194,35 @@ def append_hash_arg(symbol, *val)
     end
   end
 
+  def append_hash_array_arg(symbol, default, *val)
+    val = val.flatten.compact
+    val = val.first if val.size == 1 && val.first.kind_of?(Hash)
+    dup.tap do |r|
+      r.options[symbol] = merge_hash_or_array(r.options[symbol], val, default)
+    end
+  end
+
+  # @param a [Array, Hash]
+  # @param b [Array, Hash]
+  # @param default default value for conversion to a hash. e.g.: {} or "ASC"
+  def merge_hash_or_array(a, b, default = {})
+    if a.blank?
+      b
+    elsif b.empty?
+      a
+    elsif a.kind_of?(Array) && b.kind_of?(Array)
+      a + b
+    else
+      a = array_to_hash(a, default) if a.kind_of?(Array)
+      b = array_to_hash(b, default) if b.kind_of?(Array)
```

Comment:
Can you give examples of this?  Maybe in a comment...not sure I understand the conversion of Array to Hash.  Also, what happens if the Hash is more than one layer deep, or are we not allowing callers to do that.  On that note, does it make sense to just call deep_merge on the hashes?